packages/core/contracts/optimistic-asserter/implementation/escalation-manager/DisputeLimitingEscalationManager.sol

```diff
@@ -0,0 +1,68 @@
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./BaseEscalationManager.sol";
+import "../../interfaces/OptimisticAsserterInterface.sol";
+
+// This Escalation Manager blocks all assertions till the blocking dispute is resolved by Oracle. In order to avoid
+// interference among different applications this Escalation Manager allows assertions only from one requesting contract.
+contract DisputeLimitingEscalationManager is BaseEscalationManager, Ownable {
+    OptimisticAsserterInterface public immutable optimisticAsserter;
+
+    // Address of linked requesting contract. Before this is set via setAssertingCaller all assertions will be blocked.
+    address public assertingCaller;
+
+    bytes32 public disputedAssertionId;
+
+    event AssertingCallerSet(address indexed assertingCaller);
+
+    constructor(address _optimisticAsserter) {
+        optimisticAsserter = OptimisticAsserterInterface(_optimisticAsserter);
+    }
+
+    // Set the address of the contract that will be allowed to use Optimistic Asserter.
+    // This can only be set once. We do not set this at constructor just to allow for some flexibility in the ordering
+    // of how contracts are deployed.
+    function setAssertingCaller(address _assertingCaller) public onlyOwner {
+        require(_assertingCaller != address(0), "Invalid asserting caller");
+        require(assertingCaller == address(0), "Asserting caller already set");
+        assertingCaller = _assertingCaller;
+        emit AssertingCallerSet(_assertingCaller);
+    }
+
+    function getAssertionPolicy(bytes32 assertionId) public view override returns (AssertionPolicy memory) {
+        OptimisticAsserterInterface.Assertion memory assertion = optimisticAsserter.getAssertion(assertionId);
+        return
+            AssertionPolicy({
+                blockAssertion: _checkIfAssertionBlocked(assertion),
+                arbitrateViaEscalationManager: false,
+                discardOracle: false,
+                validateDisputers: false
+            });
+    }
+
+    function assertionDisputed(bytes32 assertionId) public override {
```

Comment:
add a comment that this is the callback.

Reply:
also looking at this code it might make sense to re-name `assertionDisputed` and `assertionResolved` to `assertionDisputedCallback` and `assertionResolvedCallback` respectively to make it clearer what these functions do in the context of the EscilationManager.